aiosmtpd/smtp.py

```diff
@@ -550,50 +582,71 @@ async def smtp_AUTH(self, arg: str) -> None:
         if await self.check_helo_needed("EHLO"):
             return
         elif not self.session.extended_smtp:
-            await self.push("500 Error: command 'AUTH' not recognized")
+            return await self.push("500 Error: command 'AUTH' not recognized")
         elif self._auth_require_tls and not self._tls_protocol:
-            await self.push("538 5.7.11 Encryption required for requested "
-                            "authentication mechanism")
+            return await self.push("538 5.7.11 Encryption required for requested "
+                                   "authentication mechanism")
         elif self.authenticated:
-            await self.push('503 Already authenticated')
+            return await self.push('503 Already authenticated')
         elif not arg:
-            await self.push('501 Not enough value')
-        else:
```

Comment:
Code is pushed to the right and honestly looks really ugly. Especially as it has gained complexity due to having to cater to the new `authenticator` system while maintaining backward compatibility with the old `auth_callback` system.

So I changed all the naked `await`s up there with `return await`